lnwire/stfu.go

```diff
@@ -0,0 +1,72 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+)
+
+// Stfu is a message that is sent to lock the channel state prior to some other
+// interactive protocol where channel updates need to be paused.
+type Stfu struct {
+	// ChanID identifies which channel needs to be frozen.
+	ChanID ChannelID
+
+	// Initiator is a byte that identifies whether we are the initiator of
+	// this process.
+	Initiator bool
+
+	// ExtraData is the set of data that was appended to this message to
+	// fill out the full maximum transport message size. These fields can
+	// be used to specify optional data such as custom TLV fields.
+	ExtraData ExtraOpaqueData
+}
+
+// A compile time check to ensure Stfu implements the lnwire.Message interface.
+var _ Message = (*Stfu)(nil)
+
+// Encode serializes the target Stfu into the passed io.Writer.
+// Serialization will observe the rules defined by the passed protocol version.
+//
+// This is a part of the lnwire.Message interface.
+func (shh *Stfu) Encode(w *bytes.Buffer, _ uint32) error {
+	var extraBytesWriter bytes.Buffer
+	shh.ExtraData = ExtraOpaqueData(extraBytesWriter.Bytes())
```

Comment:
I'd need to hear from @Roasbeef on this one. I was told to add it because doing it later would make us unable to handle TLV streams and we need to be able to tolerate TLV streams here even if we do nothing with them. I'm not 100% sure about this but that was my understanding.